### PR TITLE
Prepare Detekt 1.20.0-RC1

### PR DESCRIPTION
**Pull request by cortinico**
Time to bake a new release of Detekt 🎉 

@3flex I think we really need to switch over to having the composite build as default. Otherwise I essentially have to fix all the build failures of all the previous PRs. As an alternative, we can setup a CI job that runs a `publishToMavenLocal` and runs tests against it.

Closes #4588

### CODE REVIEW

**Comment by chao2zhang** [comment]
#4588 was a PR rather than an issue. How would `Closes #4588` work?

**Comment by cortinico** [comment]
> #4588 was a PR rather than an issue. How would `Closes #4588` work?

I've included the changes from #4588 here so as we merge this PR, the other will also be closed.

**Comment by chao2zhang** [comment]
I guess I was asking about the automation (We need to manually close #4588 after merging this PR)

File: docs/pages/changelog 1.x.x.md [hunk 1]
```diff
@@ -6,9 +6,133 @@ permalink: changelog.html
...
+  - We suppressed several warnings reported when running with type resolution - [#4423](https://github.com/detekt/detekt/pull/4423)
+- We fixed a **regression** introduced in `1.19.0` for users using `ignoreAnnotated` running **without type resolution** - [#4570](https://github.com/detekt/detekt/pull/4570)
+- We've fixed a security vulnerability related to XML parsing - [#4499](https://github.com/detekt/detekt/pull/4499)
+- We've changed the behavior of the baseline task. Now the baseline is always update, even if you fixed all the issues in your codebase - [#4445](https://github.com/detekt/detekt/pull/4445)
+- We now enable the naming ruleset by default also on tests. Previously they were excluded - [#4438](https://github.com/detekt/detekt/pull/4438)
+- This version of Detekt is built with Gradle `v7.4`, AGP `7.1.1` and Kotlin `1.6.10` (see [#4530](https://github.com/detekt/detekt/pull/4530) [#4573](https://github.com/detekt/detekt/pull/4573) [#4133](https://github.com/detekt/detekt/pull/4133) [#4277](https://github.com/detekt/detekt/pull/4277))
+- This version of Detekt is wrapping KtLint version `0.43.2` (see [#4227](https://github.com/detekt/detekt/pull/4227))
```
Comment (BraisGabin) [inline docs/pages/changelog 1.x.x.md:29]: <3 thanks for write this detailed changelog.

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**